students/Kenneth_Murray/lesson3/mailbox1/mailroom.py

```diff
@@ -0,0 +1,109 @@
+
+contributor_list = [('Ken','Murray','10.00', '4.00'),('Tew','Tangsuk','2.68','78268','265.68'),('Joe','joe','5.50','57.89'),('Tina','Tangsuk','29.02'),('Nathan','Merrill','20.00')]
+
+
+def donor_list():
+    """Returns a list of donors sorted by first name"""
+    names_of_donors = []
+    for name in contributor_list:
+        names_of_donors.append(name[0]+" "+name[1])
+    return names_of_donors
+
+def add_donor(name):
+    """Adds a first and last name to the list"""
+    contributor_list.append(tuple(name.split(" ")))
+    return contributor_list
+
+def is_donor(check_name):
+    """Return true or false"""
+    names_of_donors = []
+    for name in contributor_list:
+        names_of_donors.append(name[0]+" "+name[1])
+    for fullname in names_of_donors:
+        if fullname.casefold() == check_name.casefold():
+            return "true"
+
+
+def new_donation(donation,donor_name):
+    """Adds a new donation to the contribution list"""
+    fullname = tuple(donor_name.split(" "))
+    firstname = fullname[0]
+    lastname = fullname[1]
+    counter = 0
+    while counter <= len(contributor_list) - 1:
+        if (contributor_list[counter])[0] == firstname and (contributor_list[counter])[1] == lastname:
+            index = counter
+            contributor_list[index] = contributor_list[index] + (str(donation),)
+        counter = counter + 1
+
+
+def thankyou_email(donation,donor_name):
+    print(f"Dear {donor_name}:")
+    print()
+    print()
+    print(f"Thank you for your donation of ${donation}. Your gift is greatly appreciated.")
+    print(f"Your generosity and thoughtfulness means so much to all of us.")
+    print()
+    print()
+    print("Sincerely,")
+    print("Kenneth Murray")
+
+
+def print_report():
+    print()
+    print()
+    print('{:25} |  {:12} |   {:12} | {:13}'.format('Donor Name', 'Total Given', 'Num Gifts', 'Average Gift'))
+    print("-" * 65)
+    index = 0
+    for donor_report in contributor_list:
+        report_fullname = donor_report[0] + " " + donor_report[1]
+        report_num_gifts = int(len(donor_report) - 2)
+        report_total_given = '{:,.2f}'.format(float(donor_report[2]))
+        if report_num_gifts > 1:
+            given_counter = 3
+            while given_counter < len(donor_report):
+                report_total_given = float("{:.2f}".format(float(report_total_given))) + float("{:.2f}".format(float(donor_report[given_counter])))
+                given_counter = given_counter + 1
+        report_average = float(report_total_given) / int(report_num_gifts)
+        report_average = "{:.2f}".format(report_average)
+        report_total_given = "{:.2f}".format(float(report_total_given))
+        print(f'{report_fullname:25} | $ {report_total_given:12} |  {report_num_gifts:12} | $ {report_average:13}')
+        index = index + 1
+    print()
+    print()
+
+
+
+
+if __name__ == '__main__':
+    begin = 0
+    while begin == 0:
```

Comment:
So i think you were looking for a `while True:` statement here.  Then when you want to break the loop, you call `break` in the spot where you have `begin = 1`